sundar_grouped_bar_graph_named_paid_labor_cost_frontend

sundarmachani opened the pull request:
# Description
Implemented a new Paid Labor Cost component with an interactive grouped bar chart displaying project labor costs with comprehensive filtering capabilities.
Implements - Create Paid Labor Cost chart in the "Weekly Project Summary" (http://localhost:3000/bmdashboard/totalconstructionsummary or reports -> total construction summary)

## Related PRS (if any):
N/A

## Main changes explained:

- Created new PaidLaborCost.jsx component with grouped bar chart visualization using Chart.js
- Implemented custom PaidLaborCostDatePicker.jsx component for intuitive date range selection
- Added comprehensive namespaced CSS styles in PaidLaborCost.css with the paid-labor-cost- prefix to prevent style conflicts
- Built interactive filtering functionality for projects, tasks, and date ranges
- Implemented data aggregation logic to display appropriate metrics based on filter selections
- Added hover tooltips for detailed cost information display
- Ensured proper API communication with POST method for data retrieval


## How to test:

1. Check out the current branch
2. Run npm install and npm start to run this PR locally
3. Clear site data/cache
4. Log in as admin user
5. Go to Reports → Total Construction Summary -> Labor and Time Tracking 2nd card
6. Verify the new Paid Labor Cost component appears with the grouped bar chart
7. Test the filtering functionality:

- Select different projects from the Project filter
- Select different tasks from the Tasks filter
- Use the Date filter to select a custom date range

8. Hover over bars to verify tooltip information shows the project name, task, and cost value
9. Verify "All Projects" selection shows the total project cost and costs for the two most expensive tasks
10. Check the browser console to confirm no errors are present

## Screenshots or videos of changes:
<img width="602" alt="Screenshot 2025-04-17 at 7 22 26 PM" src="https://github.com/user-attachments/assets/f58a4079-6e30-44b0-9a8e-caaf9e7a4d81" />

<img width="604" alt="Screenshot 2025-04-17 at 7 22 41 PM" src="https://github.com/user-attachments/assets/559b484d-deaf-4679-9241-3b0ea7320eed" />


https://github.com/user-attachments/assets/ece293f5-6bc0-44f0-b758-34492bd717c0



## Note:
This PR introduces a completely new Paid Labor Cost dashboard component that meets all specified requirements:
- Grouped bar chart with project names on X-axis and costs on Y-axis
- Color-coded legend for different cost types
- Interactive filters for projects, tasks, and date ranges
- Hover functionality to display detailed cost information
- Proper handling of "All Projects" view with aggregated data
- Custom date picker implementation for intuitive date range selection
- Responsive design that adjusts to different screen sizes

## Important: 
The backend API for this component is not yet implemented. The component currently makes an API call structure but falls back to mock data when the API is unavailable. Once the backend implementation is completed, this component will be fully integrated with the actual data.

vaibhavkoladiya reviewed (dismissed):
Checked out the current branch, ran npm install and started the app. Cleared site data/cache and logged in as admin. Navigated to Reports → Total Construction Summary → Labor and Time Tracking. Verified the new Paid Labor Cost component with grouped bar chart. Tested all filters and tooltips, and confirmed correct data is shown. Approving the PR.


https://github.com/user-attachments/assets/9e90a91c-8834-4f42-abde-4659b8e0bc73

guthaVamshi reviewed (dismissed):
Checked out the current branch, ran `npm install`, and started the app. After clearing site data/cache and logging in as an admin, navigated to **Reports → Total Construction Summary → Labor and Time Tracking**. Verified the new **Paid Labor Cost** component with the grouped bar chart. Tested all filters and tooltips, and confirmed the data displays correctly. Approving this PR.

<img width="1704" alt="Screenshot 2025-04-18 at 10 27 10 AM" src="https://github.com/user-attachments/assets/9bfb5a52-1932-454a-b4f4-df0e6c586461" />

rahulprasad2823 reviewed (dismissed):
Ran the app locally and logged in as admin to test the PR. Confirmed the Paid Labor Cost component is visible under Reports → Total Construction Summary → Labor and Time Tracking. Verified that filters for project, task, and date range function correctly. Tooltips show the correct project name, task, and cost. "All Projects" displays total and top task costs accurately.  PR approved.
![Screenshot (33)](https://github.com/user-attachments/assets/158be48d-c04d-4d93-bf3c-2261e894d795)

cgomezhub reviewed (changes requested):
I have checked these functionalities:

Grouped bar chart with project names on X-axis and costs on Y-axis
Color-coded legend for different cost types
Interactive filters for projects, tasks, and date ranges
Hover functionality to display detailed cost information
Proper handling of "All Projects" view with aggregated data
Custom date picker implementation for intuitive date range selection
Responsive design that adjusts to different screen sizes.

in general this PR is working as was described, However, I recommend changing the chart bar colors in dark mode to improve viewing:


![Screenshot 2025-04-19 065706](https://github.com/user-attachments/assets/5c2f15e7-bfd0-4303-9152-61d362c24a8a)
![Screenshot 2025-04-19 065723](https://github.com/user-attachments/assets/5e90a48a-020a-48a8-9417-a821a5a5ce7f)
![Screenshot 2025-04-19 071048](https://github.com/user-attachments/assets/4343a8c8-f9df-44d2-8290-eced6fb3cbfc)

sundarmachani commented:
> I have checked these functionalities:
> 
> Grouped bar chart with project names on X-axis and costs on Y-axis Color-coded legend for different cost types Interactive filters for projects, tasks, and date ranges Hover functionality to display detailed cost information Proper handling of "All Projects" view with aggregated data Custom date picker implementation for intuitive date range selection Responsive design that adjusts to different screen sizes.
> 
> in general this PR is working as was described, However, I recommend changing the chart bar colors in dark mode to improve viewing:
> 
> ![Screenshot 2025-04-19 065706](https://private-user-images.githubusercontent.com/55204862/435388707-5c2f15e7-bfd0-4303-9152-61d362c24a8a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwOTAxMTUsIm5iZiI6MTc0NTA4OTgxNSwicGF0aCI6Ii81NTIwNDg2Mi80MzUzODg3MDctNWMyZjE1ZTctYmZkMC00MzAzLTkxNTItNjFkMzYyYzI0YThhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE5VDE5MTAxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjNDE0YjIyOTliMWU4MmI1YzVjMzM3ZTIxOTI1OTUwMjVkNGMwOGRiMGFhZDRkYjhlZGY1OWNmMjhhMmM3N2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9C1RJ-UdThhP8h1YFilo6A3VKpI6g8XnsF8TRv-RueY) ![Screenshot 2025-04-19 065723](https://private-user-images.githubusercontent.com/55204862/435388710-5e90a48a-020a-48a8-9417-a821a5a5ce7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwOTAxMTUsIm5iZiI6MTc0NTA4OTgxNSwicGF0aCI6Ii81NTIwNDg2Mi80MzUzODg3MTAtNWU5MGE0OGEtMDIwYS00OGE4LTk0MTctYTgyMWE1YTVjZTdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE5VDE5MTAxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3MDg0NDMwMTVmZTViMDMxOGE5YjE1OTViYzU3ZDkxNTdiMzhkMTkyNmE3NTZlYTljOTNmMjdkODQ3ZjJjMzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.IxELIbng8A6JDPIfb9UZUfhXEULkIzXBDgCUoDeBevk) ![Screenshot 2025-04-19 071048](https://private-user-images.githubusercontent.com/55204862/435388712-4343a8c8-f9df-44d2-8290-eced6fb3cbfc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwOTAxMTUsIm5iZiI6MTc0NTA4OTgxNSwicGF0aCI6Ii81NTIwNDg2Mi80MzUzODg3MTItNDM0M2E4YzgtZjlkZi00NGQyLTgyOTAtZWNlZDZmYjNjYmZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE5VDE5MTAxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3NDJkMWQ4NGE0NTJmYjI2ZWFkZWFjM2RlYjc0NDkyMTYxZTRhNTcyODEyNDE1YWIxMmE2NWRlY2NiMmJiMzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QosNBgUevCme98LfbLRNj5c9dGv2w240NtWcoE7BiyQ)

Hi @cgomezhub, I have changed the bar color to a different one, which appears good in both modes. Please review it.

guthaVamshi reviewed (commented):
I reviewed the PR and confirmed that I’m on the correct branch. However, the colors still appear to be the same, and I’m not seeing any changes reflected in the UI.
<img width="1697" alt="Screenshot 2025-04-21 at 12 53 09 PM" src="https://github.com/user-attachments/assets/f2163c11-3cf7-4373-8851-fa61f4ed0b00" />

sundarmachani commented:
> I reviewed the PR and confirmed that I’m on the correct branch. However, the colors still appear to be the same, and I’m not seeing any changes reflected in the UI. <img alt="Screenshot 2025-04-21 at 12 53 09 PM" width="1697" src="https://private-user-images.githubusercontent.com/79559118/435750347-f2163c11-3cf7-4373-8851-fa61f4ed0b00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUyNTUwMjAsIm5iZiI6MTc0NTI1NDcyMCwicGF0aCI6Ii83OTU1OTExOC80MzU3NTAzNDctZjIxNjNjMTEtM2NmNy00MzczLTg4NTEtZmE2MWY0ZWQwYjAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDIxVDE2NTg0MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc5MmJjNjRkMjJiMGZiZGQzNGFhZjQ4ZGFjM2I1NGQyMDA3NzAwYjI1ZmRhMDhhZjJiOTM1YWNkYmUwOTNmNzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5T_ehyn2_sh4U1f43EOnVpWPNgUNkAnCGeMcMMzB-rs">

HI @guthaVamshi thanks for the Review, could you please confirm if you took the latest pull from the branch cause I pushed some new changes lately.

gopikalakshmia reviewed (approved):
Checked out the branch and verified Labor and Time Tracking.
The following changes look good:
Grouped bar chart
Interactive filters for projects, tasks, and date ranges
Tooltip
"All Projects" functionality
Date picker
Looks good in both Dark and Light modes, including the latest Dark mode color changes.Approving this PR.
<img width="468" alt="image" src="https://github.com/user-attachments/assets/957d2d32-accb-4b5d-920f-9f6959a38e78" />

<img width="468" alt="image" src="https://github.com/user-attachments/assets/5a026229-11c1-4607-af8a-11774d6b904c" />

JuliaHa0902 reviewed (changes requested):
Hi Sundar, 
I have checked out the PR and review the functionalities of the chart. The filtering functions work well, I was able to filter by different tasks, projects and date range. When I  hover over the bar chart, I can see the details information. I tested in both dark mode and light mode, tested in mobile view, basically the chart looks good and is responsive. 
Grouped bar chart with project names on X-axis and costs on Y-axis. Color-coded legend has different colors for different cost types.
![Screenshot 2025-04-24 165453](https://github.com/user-attachments/assets/a563a099-c99f-4cab-97a5-0df76823988b)
![Screenshot 2025-04-24 165632](https://github.com/user-attachments/assets/1474e30d-1f85-48cc-beee-05283474b834)

**There are some minor issues:**
 - I have a hard time changing the start date of the custom range. I have tried to drag, click, unselect, typing but none of that works.
https://github.com/user-attachments/assets/f43cff29-efeb-4b35-9a1b-305953cf4962
 - The start date and end date color of the date range are really hard to see. In light mode, it is white text on a light blue background. In dark mode, it is black text on a dark blue background.
![Screenshot 2025-04-24 165653](https://github.com/user-attachments/assets/c3f879a2-b373-4043-ae6c-e3d798602c9e)
![Screenshot 2025-04-24 171923](https://github.com/user-attachments/assets/42925a49-c1ec-47eb-bf9f-cad96d8f079d)
 - Not sure if this is intentional or not relevant, but I can access the link http://localhost:3000/bmdashboard/totalconstructionsummary using a Volunteer account. 
![Screenshot 2025-04-24 172823](https://github.com/user-attachments/assets/b09fbb9c-99c8-4d6a-b2ec-955599e069f8)

**Suggestion:**
I see there is a requirement: "Verify "All Projects" selection shows the total project cost and costs for the **two** most expensive tasks". 
I would like to test this out, however currently now there are only 2 tasks. Can you create more mock tasks, please?

sundarmachani commented:
> Hi Sundar, I have checked out the PR and review the functionalities of the chart. The filtering functions work well, I was able to filter by different tasks, projects and date range. When I hover over the bar chart, I can see the details information. I tested in both dark mode and light mode, tested in mobile view, basically the chart looks good and is responsive. Grouped bar chart with project names on X-axis and costs on Y-axis. Color-coded legend has different colors for different cost types. ![Screenshot 2025-04-24 165453](https://private-user-images.githubusercontent.com/43044560/437230489-a563a099-c99f-4cab-97a5-0df76823988b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MTAxMDgsIm5iZiI6MTc0NTYwOTgwOCwicGF0aCI6Ii80MzA0NDU2MC80MzcyMzA0ODktYTU2M2EwOTktYzk5Zi00Y2FiLTk3YTUtMGRmNzY4MjM5ODhiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDE5MzY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYxYmM3MmI5YzU0MTIxNmM4Mzc5MjNlNzA1ZjMzNDY1MDc5NGRiZjgxNDY1OWNhNGFhZGFiZGVmMGFkZjAyNDQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.r7gGoN2ehTgJecouOyF7jIl-cba6VKgBqK1IsmLJDWI) ![Screenshot 2025-04-24 165632](https://private-user-images.githubusercontent.com/43044560/437230494-1474e30d-1f85-48cc-beee-05283474b834.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MTAxMDgsIm5iZiI6MTc0NTYwOTgwOCwicGF0aCI6Ii80MzA0NDU2MC80MzcyMzA0OTQtMTQ3NGUzMGQtMWY4NS00OGNjLWJlZWUtMDUyODM0NzRiODM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDE5MzY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRmZThjYTMxODljMzBjZDZkODJlZjY4MjRmMWEzZDk5YzJiODExZWE4ZGQyMzM1YWY4Y2NiNDJkYWFiNmI5MTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.dSPUcBxW_29Z_LF5vzlMdW4G_jxZwgQ04_SqJNIha70)
> 
> **There are some minor issues:**
> 
> * I have a hard time changing the start date of the custom range. I have tried to drag, click, unselect, typing but none of that works.
>   https://github.com/user-attachments/assets/f43cff29-efeb-4b35-9a1b-305953cf4962
> * The start date and end date color of the date range are really hard to see. In light mode, it is white text on a light blue background. In dark mode, it is black text on a dark blue background.
>   ![Screenshot 2025-04-24 165653](https://private-user-images.githubusercontent.com/43044560/437231086-c3f879a2-b373-4043-ae6c-e3d798602c9e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MTAxMDgsIm5iZiI6MTc0NTYwOTgwOCwicGF0aCI6Ii80MzA0NDU2MC80MzcyMzEwODYtYzNmODc5YTItYjM3My00MDQzLWFlNmMtZTNkNzk4NjAyYzllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDE5MzY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc2YTE3OGYyOWJkMzk2Y2RlY2Q0N2RmOTZlNmFjN2UxOGQ1NThjYzVjNjViNTA3MTQzZTM0NTJjYzljNjFjZDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._UzkM4Hqj3J6jrrMjPIONIKh7emXUHRTbNZ12MLwWo0)
>   ![Screenshot 2025-04-24 171923](https://private-user-images.githubusercontent.com/43044560/437231096-42925a49-c1ec-47eb-bf9f-cad96d8f079d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MTAxMDgsIm5iZiI6MTc0NTYwOTgwOCwicGF0aCI6Ii80MzA0NDU2MC80MzcyMzEwOTYtNDI5MjVhNDktYzFlYy00N2ViLWJmOWYtY2FkOTZkOGYwNzlkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDE5MzY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM4Y2NlZmVmNGFjZGU5NzY5ZDQ4ZTY3OTBjNzVhMGE2MWEyNGQ1N2NiNmIxMTk2Njg1YWZhNGZhNzNmODQzZTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.YA3EfpOtkHeXH4cfuQjiSBxPUcZKo3X1IZ3PvUqUHr0)
> * Not sure if this is intentional or not relevant, but I can access the link http://localhost:3000/bmdashboard/totalconstructionsummary using a Volunteer account.
>   ![Screenshot 2025-04-24 172823](https://private-user-images.githubusercontent.com/43044560/437231531-b09fbb9c-99c8-4d6a-b2ec-955599e069f8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2MTAxMDgsIm5iZiI6MTc0NTYwOTgwOCwicGF0aCI6Ii80MzA0NDU2MC80MzcyMzE1MzEtYjA5ZmJiOWMtOTljOC00ZDZhLWIyZWMtOTU1NTk5ZTA2OWY4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI1VDE5MzY0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQzYjFkYWEwZGMyNDI0YzU4NzNmNTczYzgyMmRhMWZmZTg1MWQxNTMyZjhmYzY2YzgxZWE5OWRjZTVkZmFhYjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.hg5p_gLySIg3g84D-BK_eBRkjyZni43cfGxktYwOwZk)
> 
> **Suggestion:** I see there is a requirement: "Verify "All Projects" selection shows the total project cost and costs for the **two** most expensive tasks". I would like to test this out, however currently now there are only 2 tasks. Can you create more mock tasks, please?

Hi @JuliaHa0902 I have made some additional changes for the mock data and also added an "i" button for the usage of the date picker. thanks.

JuliaHa0902 reviewed (approved):
Hi Sundar, 
I have checked out and tested the PR again. Everything works well :)
 - The information tooltip helps! I didn't notice there is a Reset button last time. The screen was cut off before the last line
 - I verified that "All Projects" selection shows the total project cost and costs for the two most expensive tasks
 - Filtering functionality works as expected. I can select different projects from the Project filter, select different tasks from the Tasks filter and use the Date filter to select a custom date range
 - Hover over bars to verify tooltip information shows the project name, task, and cost value
 
![Screenshot 2025-04-25 233716](https://github.com/user-attachments/assets/69bb9857-d38c-49e4-8255-64bc9e32ccd5)


**I have some minor suggestions though:**
 - If you could change the color of the start date and end date in the date picker, that would look perfect (Please see the red underline)
<img width="275" alt="Screenshot 2025-04-25 234429" src="https://github.com/user-attachments/assets/bf7c4b05-7fa2-4525-bc7d-8c8ae9501a66" />
<img width="262" alt="Screenshot 2025-04-25 234605" src="https://github.com/user-attachments/assets/97f306ce-095f-42c4-a4c0-9be28012b96f" />
 - Do we have a limit for the number of tasks in a project? I tested adding 20 tasks to 1 project, it doesn't work out really well. If we don't have a limit for the number of tasks in a project, maybe we should limit the number of tasks showing in the chart, or make it scrollable.
 
![Screenshot 2025-04-25 235433](https://github.com/user-attachments/assets/4907f87f-40a8-4f56-96fa-188aaabcd849)

sundarmachani commented:
> Hi Sundar, I have checked out and tested the PR again. Everything works well :)
> 
> * The information tooltip helps! I didn't notice there is a Reset button last time. The screen was cut off before the last line
> * I verified that "All Projects" selection shows the total project cost and costs for the two most expensive tasks
> * Filtering functionality works as expected. I can select different projects from the Project filter, select different tasks from the Tasks filter and use the Date filter to select a custom date range
> * Hover over bars to verify tooltip information shows the project name, task, and cost value
> 
> ![Screenshot 2025-04-25 233716](https://private-user-images.githubusercontent.com/43044560/437669559-69bb9857-d38c-49e4-8255-64bc9e32ccd5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2NDQ4MzQsIm5iZiI6MTc0NTY0NDUzNCwicGF0aCI6Ii80MzA0NDU2MC80Mzc2Njk1NTktNjliYjk4NTctZDM4Yy00OWU0LTgyNTUtNjRiYzllMzJjY2Q1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI2VDA1MTUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwMzFkYzMxYzJhMjY5MDU4ZThmYmE0NGQxZTQ0NTg1NjdiYzlhNzg2NGRkMTFmNzg4N2UyZjY1OGI1OWEyMjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PcpkSQ01qMFSUAJcCqZ6XEQIdWe2B70k0_6PU958L0M)
> 
> **I have some minor suggestions though:**
> 
> * If you could change the color of the start date and end date in the date picker, that would look perfect (Please see the red underline)
> 
> <img alt="Screenshot 2025-04-25 234429" width="275" src="https://private-user-images.githubusercontent.com/43044560/437669344-bf7c4b05-7fa2-4525-bc7d-8c8ae9501a66.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2NDQ4MzQsIm5iZiI6MTc0NTY0NDUzNCwicGF0aCI6Ii80MzA0NDU2MC80Mzc2NjkzNDQtYmY3YzRiMDUtN2ZhMi00NTI1LWJjN2QtOGM4YWU5NTAxYTY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI2VDA1MTUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmYTkzN2VhMDg2MzQwYWQzZTdiNDMzY2ZmM2Y1MGU3NTMyOTE5NTI3YjdmMGY1YjUxMTMzNmY2M2QwNzc5ZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.iYdxhC0Mac0MzsVts9I4c-V9pfcG1gEArcWCaTmFdlI"> <img alt="Screenshot 2025-04-25 234605" width="262" src="https://private-user-images.githubusercontent.com/43044560/437669346-97f306ce-095f-42c4-a4c0-9be28012b96f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2NDQ4MzQsIm5iZiI6MTc0NTY0NDUzNCwicGF0aCI6Ii80MzA0NDU2MC80Mzc2NjkzNDYtOTdmMzA2Y2UtMDk1Zi00MmM0LWE0YzAtOWJlMjgwMTJiOTZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI2VDA1MTUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNhYTBhMTlhNWMwZWVhOWVjZGI4Mzc1ZjQ5NmNhMjBhZWE3NjllZWUzMzk5YzY1YjM0ZWUyZGU4OTMyMmNkNTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VRT_ZSB5kd9feXdYPYBlhkkXkz0A9X_Ts_uA3Q10puE"> - Do we have a limit for the number of tasks in a project? I tested adding 20 tasks to 1 project, it doesn't work out really well. If we don't have a limit for the number of tasks in a project, maybe we should limit the number of tasks showing in the chart, or make it scrollable.
> ![Screenshot 2025-04-25 235433](https://private-user-images.githubusercontent.com/43044560/437669483-4907f87f-40a8-4f56-96fa-188aaabcd849.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2NDQ4MzQsIm5iZiI6MTc0NTY0NDUzNCwicGF0aCI6Ii80MzA0NDU2MC80Mzc2Njk0ODMtNDkwN2Y4N2YtNDBhOC00ZjU2LTk2ZmEtMTg4YWFhYmNkODQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI2VDA1MTUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiYjRlYmEzZmU4MjdjNTJhM2JjNmY2ZjQ1ODRlMGQ5OTA4ODgyNmQ1NjJkOTA0ODEzMTRlYjg2YWJmY2RjNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5w3X6Ve-uZ-f2y3WCkMUloytvhRg4LPAHa8SUg7l7uM)

Hi @JuliaHa0902,
Thank you for your review! Since there weren’t any explicit requirements around task-display limits, I haven’t made changes in that area yet. Would it be possible for you to gather all of your observations and suggestions into a single pass? That would help me consolidate updates in fewer commits and keep the PR history clean. Thanks so much for your help!

cgomezhub commented:
> > I have checked these functionalities:
> > Grouped bar chart with project names on X-axis and costs on Y-axis Color-coded legend for different cost types Interactive filters for projects, tasks, and date ranges Hover functionality to display detailed cost information Proper handling of "All Projects" view with aggregated data Custom date picker implementation for intuitive date range selection Responsive design that adjusts to different screen sizes.
> > in general this PR is working as was described, However, I recommend changing the chart bar colors in dark mode to improve viewing:
> > ![Screenshot 2025-04-19 065706](https://private-user-images.githubusercontent.com/55204862/435388707-5c2f15e7-bfd0-4303-9152-61d362c24a8a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwOTAxMTUsIm5iZiI6MTc0NTA4OTgxNSwicGF0aCI6Ii81NTIwNDg2Mi80MzUzODg3MDctNWMyZjE1ZTctYmZkMC00MzAzLTkxNTItNjFkMzYyYzI0YThhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE5VDE5MTAxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThjNDE0YjIyOTliMWU4MmI1YzVjMzM3ZTIxOTI1OTUwMjVkNGMwOGRiMGFhZDRkYjhlZGY1OWNmMjhhMmM3N2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.9C1RJ-UdThhP8h1YFilo6A3VKpI6g8XnsF8TRv-RueY) ![Screenshot 2025-04-19 065723](https://private-user-images.githubusercontent.com/55204862/435388710-5e90a48a-020a-48a8-9417-a821a5a5ce7f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwOTAxMTUsIm5iZiI6MTc0NTA4OTgxNSwicGF0aCI6Ii81NTIwNDg2Mi80MzUzODg3MTAtNWU5MGE0OGEtMDIwYS00OGE4LTk0MTctYTgyMWE1YTVjZTdmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE5VDE5MTAxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk3MDg0NDMwMTVmZTViMDMxOGE5YjE1OTViYzU3ZDkxNTdiMzhkMTkyNmE3NTZlYTljOTNmMjdkODQ3ZjJjMzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.IxELIbng8A6JDPIfb9UZUfhXEULkIzXBDgCUoDeBevk) ![Screenshot 2025-04-19 071048](https://private-user-images.githubusercontent.com/55204862/435388712-4343a8c8-f9df-44d2-8290-eced6fb3cbfc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDUwOTAxMTUsIm5iZiI6MTc0NTA4OTgxNSwicGF0aCI6Ii81NTIwNDg2Mi80MzUzODg3MTItNDM0M2E4YzgtZjlkZi00NGQyLTgyOTAtZWNlZDZmYjNjYmZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDE5VDE5MTAxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ3NDJkMWQ4NGE0NTJmYjI2ZWFkZWFjM2RlYjc0NDkyMTYxZTRhNTcyODEyNDE1YWIxMmE2NWRlY2NiMmJiMzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QosNBgUevCme98LfbLRNj5c9dGv2w240NtWcoE7BiyQ)
> 
> Hi @cgomezhub, I have changed the bar color to a different one, which appears good in both modes. Please review it.

Perfect!, changes are applied,

![Screenshot 2025-04-26 064133](https://github.com/user-attachments/assets/9317905e-0d87-449c-9459-6a7b999c9088)

cgomezhub reviewed (approved):
I have checked these functionalities:

Grouped bar chart with project names on X-axis and costs on Y-axis
Color-coded legend for different cost types
Interactive filters for projects, tasks, and date ranges
Hover functionality to display detailed cost information
Proper handling of "All Projects" view with aggregated data
Custom date picker implementation for intuitive date range selection
Responsive design that adjusts to different screen sizes.

in general this PR is working as was described, However, changes  in the chart bar colors in dark mode to improve viewing are applied.
![Screenshot 2025-04-26 064133](https://github.com/user-attachments/assets/95184996-b4e1-47c8-856b-da9de5722b06)

sundarmachani commented:
> > Hi Sundar, I have checked out and tested the PR again. Everything works well :)
> > 
> > * The information tooltip helps! I didn't notice there was a Reset button last time. The screen was cut off before the last line
> > * I verified that "All Projects" selection shows the total project cost and costs for the two most expensive tasks
> > * Filtering functionality works as expected. I can select different projects from the Project filter, select different tasks from the Tasks filter and use the Date filter to select a custom date range
> > * Hover over bars to verify tooltip information shows the project name, task, and cost value
> > 
> > ![Screenshot 2025-04-25 233716](https://private-user-images.githubusercontent.com/43044560/437669559-69bb9857-d38c-49e4-8255-64bc9e32ccd5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2NDQ4MzQsIm5iZiI6MTc0NTY0NDUzNCwicGF0aCI6Ii80MzA0NDU2MC80Mzc2Njk1NTktNjliYjk4NTctZDM4Yy00OWU0LTgyNTUtNjRiYzllMzJjY2Q1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI2VDA1MTUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgwMzFkYzMxYzJhMjY5MDU4ZThmYmE0NGQxZTQ0NTg1NjdiYzlhNzg2NGRkMTFmNzg4N2UyZjY1OGI1OWEyMjImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PcpkSQ01qMFSUAJcCqZ6XEQIdWe2B70k0_6PU958L0M)
> > **I have some minor suggestions though:**
> > 
> > * If you could change the color of the start date and end date in the date picker, that would look perfect (Please see the red underline)
> > 
> > <img alt="Screenshot 2025-04-25 234429" width="275" src="https://private-user-images.githubusercontent.com/43044560/437669344-bf7c4b05-7fa2-4525-bc7d-8c8ae9501a66.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2NDQ4MzQsIm5iZiI6MTc0NTY0NDUzNCwicGF0aCI6Ii80MzA0NDU2MC80Mzc2NjkzNDQtYmY3YzRiMDUtN2ZhMi00NTI1LWJjN2QtOGM4YWU5NTAxYTY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI2VDA1MTUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdmYTkzN2VhMDg2MzQwYWQzZTdiNDMzY2ZmM2Y1MGU3NTMyOTE5NTI3YjdmMGY1YjUxMTMzNmY2M2QwNzc5ZjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.iYdxhC0Mac0MzsVts9I4c-V9pfcG1gEArcWCaTmFdlI"> <img alt="Screenshot 2025-04-25 234605" width="262" src="https://private-user-images.githubusercontent.com/43044560/437669346-97f306ce-095f-42c4-a4c0-9be28012b96f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2NDQ4MzQsIm5iZiI6MTc0NTY0NDUzNCwicGF0aCI6Ii80MzA0NDU2MC80Mzc2NjkzNDYtOTdmMzA2Y2UtMDk1Zi00MmM0LWE0YzAtOWJlMjgwMTJiOTZmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI2VDA1MTUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNhYTBhMTlhNWMwZWVhOWVjZGI4Mzc1ZjQ5NmNhMjBhZWE3NjllZWUzMzk5YzY1YjM0ZWUyZGU4OTMyMmNkNTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VRT_ZSB5kd9feXdYPYBlhkkXkz0A9X_Ts_uA3Q10puE"> - Do we have a limit for the number of tasks in a project? I tested adding 20 tasks to 1 project, it doesn't work out really well. If we don't have a limit for the number of tasks in a project, maybe we should limit the number of tasks showing in the chart, or make it scrollable.
> > ![Screenshot 2025-04-25 235433](https://private-user-images.githubusercontent.com/43044560/437669483-4907f87f-40a8-4f56-96fa-188aaabcd849.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDU2NDQ4MzQsIm5iZiI6MTc0NTY0NDUzNCwicGF0aCI6Ii80MzA0NDU2MC80Mzc2Njk0ODMtNDkwN2Y4N2YtNDBhOC00ZjU2LTk2ZmEtMTg4YWFhYmNkODQ5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDI2VDA1MTUzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNiYjRlYmEzZmU4MjdjNTJhM2JjNmY2ZjQ1ODRlMGQ5OTA4ODgyNmQ1NjJkOTA0ODEzMTRlYjg2YWJmY2RjNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.5w3X6Ve-uZ-f2y3WCkMUloytvhRg4LPAHa8SUg7l7uM)
> 
> Hi @JuliaHa0902, Thank you for your review! Since there weren’t any explicit requirements around task-display limits, I haven’t made changes in that area yet. Would it be possible for you to gather all of your observations and suggestions into a single pass? That would help me consolidate updates in fewer commits and keep the PR history clean. Thanks so much for your help!

Hi @JuliaHa0902, I have made the improvements so that you can select or deselect the start and end dates without reset, and also I have changed the text color of the starDate and EndDate. And also added a scroll bar, hope you like it.

JuliaHa0902 reviewed (approved):
Everything works beautifully. The color of start date and end date is fixed. The graph can handle a large amount of task data. I can select different date range by clicking again on the date picker. I approved. Thank you!
Sorry I didn't mean to drag the suggestions in multiple commits, I just didn't think of and realize all the problems when I first reviewed the PR. Sorry for any inconvenience.
![Screenshot 2025-04-28 221336](https://github.com/user-attachments/assets/4a3703d6-db0c-4064-ab4a-0b2f34b85fcf)
![Screenshot 2025-04-28 221359](https://github.com/user-attachments/assets/02bedcb3-88c0-40d6-aba0-f573cd169c97)
![Screenshot 2025-04-28 221652](https://github.com/user-attachments/assets/a73b24cd-ae7e-420c-a062-34816ca801f0)

cgomezhub reviewed (approved):
now it is possible to pick differents dates and data colors are displayed correctly.
good job!

![Screenshot 2025-05-04 080325](https://github.com/user-attachments/assets/e0074896-8d44-49d6-b736-36729731ff8d)
![Screenshot 2025-05-04 080418](https://github.com/user-attachments/assets/5019be4e-c276-4a60-8b52-07d410de38f1)

npolasani reviewed (approved):
Checked out the branch locally and ran the application.
Logged in as an admin and navigated to: Reports → Total Construction Summary.
Tested the grouped bar chart by selecting various projects, tasks, and date filters.
Verified that the bar graph displays correctly in dark mode.
All functionalities are working as expected.

Approving the PR.

<img width="464" alt="image" src="https://github.com/user-attachments/assets/eb9a0047-a461-4e5c-83b9-ecb49986420f" />
